lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/AzureStorageClient.java

```diff
@@ -20,120 +20,131 @@
 import java.io.InputStream;
 import java.net.URI;
 import java.time.Duration;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Map;
-import java.util.regex.Pattern;
 import org.apache.commons.codec.digest.DigestUtils;
 import org.apache.commons.io.FilenameUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class AzureStorageClient extends BaseStorageClient {
 
-  private BlobContainerClient containerClient;
+  private BlobServiceClient serviceClient;
   private static final Logger log = LoggerFactory.getLogger(AzureStorageClient.class);
 
-  public AzureStorageClient(URI pathToStorage, String docIdPrefix, List<Pattern> excludes, List<Pattern> includes,
-      Map<String, Object> cloudOptions, Config fileOptions) {
-    super(pathToStorage, docIdPrefix, excludes, includes, cloudOptions, fileOptions);
+  public AzureStorageClient(Config azureCloudOptions) {
+    super(azureCloudOptions);
   }
 
   @Override
-  protected String getContainerOrBucketName() {
-    return pathToStorageURI.getPath().split("/")[1];
-  }
-
-  @Override
-  protected String getStartingDirectory() {
-    String path = pathToStorageURI.getPath();
-    // path is in the format /containerName/folder1/folder2/... so need to return folder1/folder2/...
-    String[] subPaths = path.split("/", 3);
-    return subPaths.length > 2 ? subPaths[2] : "";
+  protected void validateOptions(Config cloudOptions) {
+    if (!validOptions(cloudOptions)) {
+      throw new IllegalArgumentException("Either '" + AZURE_CONNECTION_STRING + "' or '" + AZURE_ACCOUNT_NAME + "' & '" + AZURE_ACCOUNT_KEY + "' has to be in cloudOptions for AzureStorageClient.");
```

Comment:
exception message still references cloudOptions but we took that out -- should update in accordance with the new config format